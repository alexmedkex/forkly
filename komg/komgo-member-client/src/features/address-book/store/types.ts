import { List } from 'immutable'
import { ImmutableMap } from '../../../utils/types'
import { Action } from 'redux'
import { IMember } from '../../members/store/types'

export enum AddressBookActionType {
  GET_COMPANIES_REQUEST = '@@addressBook/GET_COMPANIESS_REQUEST',
  GET_COMPANIES_SUCCESS = '@@addressBook/GET_COMPANIESS_SUCCESS',
  GET_COMPANIES_FAILURE = '@@addressBook/GET_COMPANIESS_FAILURE',
  GET_COMPANY_REQUEST = '@@addressBook/GET_COMPANY_REQUEST',
  GET_COMPANY_SUCCESS = '@@addressBook/GET_COMPANY_SUCCESS',
  GET_COMPANY_FAILURE = '@@addressBook/GET_COMPANY_FAILURE',
  GENERATE_MEMBER_PACKAGE_REQUEST = '@@addressBook/GENERATE_MEMBER_PACKAGE_REQUEST',
  GENERATE_MEMBER_PACKAGE_SUCCESS = '@@addressBook/GENERATE_MEMBER_PACKAGE_SUCCESS',
  GENERATE_MEMBER_PACKAGE_FAILURE = '@@addressBook/GENERATE_MEMBER_PACKAGE_FAILURE',
  ADD_COMPANY_TO_ENS_REQUEST = '@@addressBook/ADD_COMPANY_TO_ENS_REQUEST',
  ADD_COMPANY_TO_ENS_SUCCESS = '@@addressBook/ADD_COMPANY_TO_ENS_SUCCESS',
  ADD_COMPANY_TO_ENS_FAILURE = '@@addressBook/ADD_COMPANY_TO_ENS_FAILURE',
  CONFIGURE_MQ_REQUEST = '@@addressBook/CONFIGURE_MQ_REQUEST',
  CONFIGURE_MQ_SUCCESS = '@@addressBook/CONFIGURE_MQ_SUCCESS',
  CONFIGURE_MQ_FAILURE = '@@addressBook/CONFIGURE_MQ_FAILURE',
  CREATE_COMPANY_REQUEST = '@@addressBook/CREATE_COMPANY_REQUEST',
  CREATE_COMPANY_SUCCESS = '@@addressBook/CREATE_COMPANY_SUCCESS',
  CREATE_COMPANY_FAILURE = '@@addressBook/CREATE_COMPANY_FAILURE',
  UPDATE_COMPANY_REQUEST = '@@addressBook/UPDATE_COMPANY_REQUEST',
  UPDATE_COMPANY_SUCCESS = '@@addressBook/UPDATE_COMPANY_SUCCESS',
  UPDATE_COMPANY_FAILURE = '@@addressBook/UPDATE_COMPANY_FAILURE',
  DeleteCompanyByIdSuccess = '@@addressBook/DELETE_COMPANY_BY_ID_SUCCESS'
}

export interface AddressBookStateProperties {
  companies: List<IMember>
}

export type AddressBookState = ImmutableMap<AddressBookStateProperties>

export interface GetCompaniesRequest extends Action {
  type: AddressBookActionType.GET_COMPANIES_SUCCESS
  payload: IMember[]
}
export interface GetCompanyRequest extends Action {
  type: AddressBookActionType.GET_COMPANY_SUCCESS
  payload: IMember
}
export interface GenerateMemberPackageRequest extends Action {
  type: AddressBookActionType.GENERATE_MEMBER_PACKAGE_SUCCESS
  payload: IMember
}
export interface AddCompanyToENSRequest extends Action {
  type: AddressBookActionType.ADD_COMPANY_TO_ENS_SUCCESS
  payload: IMember
}
export interface ConfigureMQRequest extends Action {
  type: AddressBookActionType.CONFIGURE_MQ_SUCCESS
  payload: IMember
}

export interface CreateCompanyRequest extends Action {
  type: AddressBookActionType.CREATE_COMPANY_SUCCESS
  payload: IMember
}
export interface UpdateCompanyRequest extends Action {
  type: AddressBookActionType.UPDATE_COMPANY_SUCCESS
  payload: IMember
}

export interface DeleteCompanyByIdSuccess extends Action {
  type: AddressBookActionType.DeleteCompanyByIdSuccess
  payload: { id: string }
}

export type AddressBookActions =
  | GetCompaniesRequest
  | GetCompanyRequest
  | CreateCompanyRequest
  | UpdateCompanyRequest
  | GenerateMemberPackageRequest
  | AddCompanyToENSRequest
  | ConfigureMQRequest
  | DeleteCompanyByIdSuccess
