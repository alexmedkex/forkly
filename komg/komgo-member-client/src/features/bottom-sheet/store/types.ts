import { Action } from 'redux'
import { ImmutableMap } from '../../../utils/types'
import {
  CreateDocumentError,
  CreateDocumentSuccess,
  DocumentRegisteredError,
  DocumentRegisteredSuccess,
  HasId,
  HasName
} from '../../document-management/store/types'
import { ActionCreator } from 'react-redux'

export enum BottomSheetActionType {
  REMOVE_BOTTOMSHEET_ITEM = '@@btsh/REMOVE_BOTTOMSHEET_ITEM',
  RETRY_BOTTOMSHEET_ITEM = '@@btsh/RETRY_BOTTOMSHEET_ITEM',
  UPDATE_BOTTOMSHEET_ITEM = '@@btsh/UPDATE_BOTTOMSHEET_ITEM',
  CREATE_BOTTOMSHEET_ITEM = '@@btsh/CREATE_BOTTOMSHEET_ITEM'
}

export interface BottomSheetItem extends HasId, HasName {
  state: BottomSheetStatus
  itemType?: BottomSheetItemType
  displayStatus?: string
  action?: ActionCreator<any>
  navigationLink?: string
}

export enum BottomSheetItemType {
  REGISTER_KYC_DOCUMENT,
  CREATE_MAGIC_LINK,
  REGISTER_RD_DOCUMENT,
  REGISTER_TRADE_DOCUMENT
}

export enum BottomSheetStatus {
  NEW = 'NEW',
  PENDING = 'PENDING',
  REGISTERED = 'REGISTERED',
  FAILED = 'FAILED'
}

export interface BottomSheetStateFields {
  visible: boolean
  items: BottomSheetItem[]
}

export interface RemoveBottomSheetItem extends Action {
  type: BottomSheetActionType.REMOVE_BOTTOMSHEET_ITEM
  payload: HasId
}

export interface RetryRegisterItem extends Action {
  type: BottomSheetActionType.RETRY_BOTTOMSHEET_ITEM
  payload: BottomSheetItem
}

export interface UpdateBottomSheetItem extends Action {
  type: BottomSheetActionType.UPDATE_BOTTOMSHEET_ITEM
  payload: BottomSheetItem
}

export interface CreateBottomSheetItem extends Action {
  type: BottomSheetActionType.CREATE_BOTTOMSHEET_ITEM
  payload: BottomSheetItem
}

export type BottomSheetState = ImmutableMap<BottomSheetStateFields>

export type BottomSheetAction =
  | DocumentRegisteredSuccess
  | DocumentRegisteredError
  | CreateDocumentSuccess
  | CreateDocumentError
  | RemoveBottomSheetItem
  | RetryRegisterItem
  | UpdateBottomSheetItem
  | CreateBottomSheetItem
