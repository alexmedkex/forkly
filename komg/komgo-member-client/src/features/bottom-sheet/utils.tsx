import { BottomSheetItem, BottomSheetStatus, BottomSheetAction, BottomSheetItemType } from './store/types'
import React from 'react'

export const generateBottomsheetTitle = (items: BottomSheetItem[]) => {
  let textHeader = ''
  if (items[0]) {
    const latestStatus = items[0].state
    const count = items.filter(x => x.state === latestStatus).length
    textHeader = getPrintableStatus(latestStatus, count)
  }
  return textHeader
}

const getPrintableStatus = (latestStatus: BottomSheetStatus, count: number) => {
  switch (latestStatus) {
    case BottomSheetStatus.REGISTERED:
      return `${count} document${count > 1 ? 's' : ''} registered`
    case BottomSheetStatus.PENDING:
      return `${count} document${count > 1 ? 's' : ''} pending`
    case BottomSheetStatus.FAILED:
      return `${count} failed registration`
  }
}

export const getStatus = (action: BottomSheetAction): BottomSheetStatus => {
  return BottomSheetStatus.PENDING
}

export function fakeBottomsheetItem<T extends BottomSheetItem>(partial?: Partial<T>): BottomSheetItem {
  const { id, state, name, itemType } = {
    ...{
      id: '-1',
      name: 'anon',
      state: BottomSheetStatus.PENDING,
      itemType: BottomSheetItemType.REGISTER_KYC_DOCUMENT
    },
    ...partial
  }
  return { id, state, name, itemType }
}
