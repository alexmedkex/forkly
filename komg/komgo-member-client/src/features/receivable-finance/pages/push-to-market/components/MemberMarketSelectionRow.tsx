import * as React from 'react'
import { Table, Popup, Checkbox } from 'semantic-ui-react'
import { Counterparty } from '../../../../counterparties/store/types'
import { TruncatedText } from '../../../../../components'
import Text from '../../../../../components/text'
import { IMemberMarketSelectionItem } from '../../../../receivable-discounting-legacy/store/types'

export interface MemberMarketSelectionRowProps {
  data: IMemberMarketSelectionItem
  selectedData: Counterparty[]
  handleCheckboxClick(counterparty?: Counterparty): void
}

const DataCell: React.FC<{ children: any }> = ({ children }) => (
  <span style={{ paddingTop: '8px' }}>
    <Text>
      <TruncatedText text={children} maxLength={20} />
    </Text>
  </span>
)

const MemberMarketSelectionRow: React.FC<MemberMarketSelectionRowProps> = (props: MemberMarketSelectionRowProps) => {
  const {
    data: {
      counterparty: {
        staticId,
        x500Name: { STREET, L, PC, CN }
      }
    },
    data,
    handleCheckboxClick,
    selectedData
  } = props
  return (
    <Table.Row id={staticId}>
      <Table.Cell id={`${staticId}-checkbox`} data-test-id={`${staticId}-checkbox`} textAlign={'left'}>
        <Checkbox
          onClick={() => handleCheckboxClick(data.counterparty)}
          checked={selectedData.includes(data.counterparty)}
        />
      </Table.Cell>
      <Table.Cell id={`${staticId}-source`} data-test-id={`${staticId}-source`}>
        <Popup
          inverted={true}
          position={'right center'}
          trigger={
            <span style={{ paddingTop: '8px' }}>
              <Text bold={true}>
                <TruncatedText text={CN} maxLength={20} />
              </Text>
            </span>
          }
          content={[CN, STREET, L, PC].join('\n')}
        />
      </Table.Cell>
      <Table.Cell data-test-id={`${staticId}-location`}>
        <DataCell>{data.location}</DataCell>
      </Table.Cell>
      <Table.Cell data-test-id={`${staticId}-appetite`}>
        <DataCell>{data.appetite}</DataCell>
      </Table.Cell>
      <Table.Cell data-test-id={`${staticId}-availability`}>
        <DataCell>{data.availability}</DataCell>
      </Table.Cell>
      <Table.Cell data-test-id={`${staticId}-creditLimit`}>
        <DataCell>{data.creditLimit}</DataCell>
      </Table.Cell>
      <Table.Cell data-test-id={`${staticId}-riskFee`}>
        <DataCell>{data.riskFee}</DataCell>
      </Table.Cell>
      <Table.Cell data-test-id={`${staticId}-margin`}>
        <DataCell>{data.margin}</DataCell>
      </Table.Cell>
      <Table.Cell data-test-id={`${staticId}-maxTenor`}>
        <DataCell>{data.maxTenor}</DataCell>
      </Table.Cell>
    </Table.Row>
  )
}

export default MemberMarketSelectionRow
