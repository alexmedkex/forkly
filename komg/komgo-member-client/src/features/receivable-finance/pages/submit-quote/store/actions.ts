import { ActionCreator } from 'react-redux'
import { Action } from 'redux'

import {
  ReceivableDiscountingActionThunk,
  fetchDiscountingRequest
} from '../../../../receivable-discounting-legacy/store/application/actions'
import { fetchCreditLines } from '../../../../credit-line/store/actions'
import { Products } from '../../../../document-management/constants/Products'
import { SubProducts } from '../../../../document-management/constants/SubProducts'

export const fetchRDRequesForSubmitQuote: ActionCreator<ReceivableDiscountingActionThunk> = (rdId: string) => (
  dispatch,
  _,
  api
): Action => {
  fetchCreditLines(Products.TradeFinance, SubProducts.ReceivableDiscounting)(dispatch, _ as any, api)
  return fetchDiscountingRequest(rdId)(dispatch, _, api)
}
