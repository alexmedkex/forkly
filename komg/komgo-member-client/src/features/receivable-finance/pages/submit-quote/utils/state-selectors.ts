import { CreditLineType, IExtendedCreditLine } from '../../../../credit-line/store/types'
import { getExtendedCreditLineArray } from '../../../../credit-line/utils/state-selectors'
import { IReceivablesDiscountingInfo, ITrade } from '@komgo/types'
import { getReceivableDiscountingInfoByRdId } from '../../../../receivable-discounting-legacy/utils/state-selectors'

export const getRDExtendedCreditLine = (state, rdId: string): IExtendedCreditLine => {
  const extendedCreditLines: IExtendedCreditLine[] = getExtendedCreditLineArray(state, CreditLineType.RiskCover)
  const rdInfo: IReceivablesDiscountingInfo = getReceivableDiscountingInfoByRdId(state, rdId)
  if (rdInfo && extendedCreditLines && extendedCreditLines.length > 0) {
    const filteredArray = extendedCreditLines.filter(
      (line: IExtendedCreditLine) => line.counterpartyStaticId === rdInfo.tradeSnapshot.trade.buyer
    )
    return filteredArray.length > 0 ? filteredArray[0] : undefined
  }
  return undefined
}
