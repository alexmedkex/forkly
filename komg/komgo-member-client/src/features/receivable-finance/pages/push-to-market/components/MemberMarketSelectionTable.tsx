import * as React from 'react'
import { Table, Checkbox } from 'semantic-ui-react'
import { productRD } from '@komgo/products'
import { Counterparty } from '../../../../counterparties/store/types'
import MemberMarketSelectionRow from './MemberMarketSelectionRow'
import { WithLicenseCheckProps } from '../../../../../components'
import { IMemberMarketSelectionItem } from '../../../../receivable-discounting-legacy/store/types'

export interface IMemberMarketSelectionTableProps extends WithLicenseCheckProps {
  data: IMemberMarketSelectionItem[]
  selectedData: Counterparty[]
  handleCheckboxClick(counterparty?: Counterparty): void
}

export class MemberMarketSelectionTable extends React.Component<IMemberMarketSelectionTableProps> {
  constructor(props: IMemberMarketSelectionTableProps) {
    super(props)
  }

  render() {
    const { data, handleCheckboxClick, selectedData, isLicenseEnabledForCompany } = this.props

    return (
      <Table basic="very" textAlign="left" sortable={true}>
        <Table.Header>
          <Table.Row>
            <Table.HeaderCell id="select" textAlign={'left'} width={'1'} style={{ padding: '14px 14px 5px 14px' }}>
              <Checkbox data-test-id="checkbox-select-all-none" onClick={() => handleCheckboxClick()} />
            </Table.HeaderCell>
            <Table.HeaderCell id="provider" textAlign={'left'}>
              Provider
            </Table.HeaderCell>
            <Table.HeaderCell id="location" textAlign={'left'}>
              Location
            </Table.HeaderCell>
            <Table.HeaderCell id="appetite" textAlign={'left'}>
              Appetite
            </Table.HeaderCell>
            <Table.HeaderCell id="availability" textAlign={'left'}>
              Availability
            </Table.HeaderCell>
            <Table.HeaderCell id="creditLimit" textAlign={'left'}>
              Credit Limit
            </Table.HeaderCell>
            <Table.HeaderCell id="riskFee" textAlign={'left'}>
              Risk Fee
            </Table.HeaderCell>
            <Table.HeaderCell id="margin" textAlign={'left'}>
              Margin
            </Table.HeaderCell>
            <Table.HeaderCell id="maxTenor" textAlign={'left'}>
              Max Tenor (Days)
            </Table.HeaderCell>
          </Table.Row>
        </Table.Header>
        <Table.Body>
          {(data as IMemberMarketSelectionItem[]).reduce((res, cp, idx) => {
            const row = (
              <MemberMarketSelectionRow
                key={idx}
                data={cp}
                selectedData={selectedData}
                handleCheckboxClick={handleCheckboxClick}
              />
            )
            return isLicenseEnabledForCompany(productRD, cp.counterparty.staticId) ? [...res, row] : res
          }, [])}
        </Table.Body>
      </Table>
    )
  }
}

export default MemberMarketSelectionTable
