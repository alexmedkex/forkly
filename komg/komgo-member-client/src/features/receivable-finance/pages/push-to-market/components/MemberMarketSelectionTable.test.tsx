import * as React from 'react'
import { shallow } from 'enzyme'
import * as renderer from 'react-test-renderer'

import MemberMarketSelectionTable, { IMemberMarketSelectionTableProps } from './MemberMarketSelectionTable'
import { Counterparty } from '../../../../counterparties/store/types'
import { fakeCounterparty } from '../../../../letter-of-credit-legacy/utils/faker'
import MemberMarketSelectionRow from './MemberMarketSelectionRow'
import { TableBody } from 'semantic-ui-react'
import { IMemberMarketSelectionItem } from '../../../../receivable-discounting-legacy/store/types'
import { fakeMemberMarketSelectionItem } from '../../../../receivable-discounting-legacy/utils/faker'

const counterparties: Counterparty[] = [
  fakeCounterparty({ staticId: '1', commonName: 'A Company', isFinancialInstitution: true, isMember: true }),
  fakeCounterparty({ staticId: '2', commonName: 'B Company', isFinancialInstitution: true, isMember: true }),
  fakeCounterparty({ staticId: '3', commonName: 'C Company', isFinancialInstitution: true, isMember: false })
]

const data: IMemberMarketSelectionItem[] = [
  fakeMemberMarketSelectionItem({ counterparty: counterparties[0] }),
  fakeMemberMarketSelectionItem({ counterparty: counterparties[1] }),
  fakeMemberMarketSelectionItem({ counterparty: counterparties[2] })
]

const mockIsLicenseEnabledForCompany = jest.fn((_, staticId) => true)
const testProps: IMemberMarketSelectionTableProps = {
  data,
  selectedData: [],
  isLicenseEnabled: jest.fn(() => true),
  isLicenseEnabledForCompany: mockIsLicenseEnabledForCompany,
  handleCheckboxClick: jest.fn((counterparty?: Counterparty) => void 0)
}

it('renders correctly', () => {
  expect(renderer.create(<MemberMarketSelectionTable {...testProps} />).toJSON()).toMatchSnapshot()
})

it('calls handleCheckboxClick when select all/none checkbox clicked', () => {
  const wrapper = shallow(<MemberMarketSelectionTable {...testProps} />)

  const selectAllNoneCheckbox = wrapper.find({ 'data-test-id': 'checkbox-select-all-none' }).shallow()
  selectAllNoneCheckbox.simulate('click')
  expect(testProps.handleCheckboxClick).toBeCalled()
})

it('shows 2 counterparty rows', () => {
  const wrapper = shallow(<MemberMarketSelectionTable {...testProps} />)

  const counterpartyTableRows = wrapper
    .find(TableBody)
    .shallow()
    .find(MemberMarketSelectionRow)

  expect(counterpartyTableRows.length === 2)
})

it('shows 1 counterparty row if only one counterparty has RD license', () => {
  mockIsLicenseEnabledForCompany.mockImplementation((_, staticId) => staticId === 2)
  const wrapper = shallow(<MemberMarketSelectionTable {...testProps} />)

  const counterpartyTableRows = wrapper
    .find(TableBody)
    .shallow()
    .find(MemberMarketSelectionRow)

  expect(counterpartyTableRows.length === 1)
})
