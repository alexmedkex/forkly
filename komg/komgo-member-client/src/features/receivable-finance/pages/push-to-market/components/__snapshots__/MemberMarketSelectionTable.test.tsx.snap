// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  display: inline-block;
  font-size: 14px;
  font-weight: bold;
  color: #000000;
  margin: 0;
}

.c1 {
  display: inline-block;
  font-size: 14px;
  font-weight: normal;
  color: #000000;
  margin: 0;
}

<table
  className="ui sortable very basic left aligned table"
>
  <thead
    className=""
  >
    <tr
      className=""
    >
      <th
        className="left aligned one wide"
        id="select"
        style={
          Object {
            "padding": "14px 14px 5px 14px",
          }
        }
      >
        <div
          className="ui fitted checkbox"
          data-test-id="checkbox-select-all-none"
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <input
            checked={false}
            className="hidden"
            onClick={[Function]}
            readOnly={true}
            tabIndex={0}
            type="checkbox"
          />
          <label />
        </div>
      </th>
      <th
        className="left aligned"
        id="provider"
      >
        Provider
      </th>
      <th
        className="left aligned"
        id="location"
      >
        Location
      </th>
      <th
        className="left aligned"
        id="appetite"
      >
        Appetite
      </th>
      <th
        className="left aligned"
        id="availability"
      >
        Availability
      </th>
      <th
        className="left aligned"
        id="creditLimit"
      >
        Credit Limit
      </th>
      <th
        className="left aligned"
        id="riskFee"
      >
        Risk Fee
      </th>
      <th
        className="left aligned"
        id="margin"
      >
        Margin
      </th>
      <th
        className="left aligned"
        id="maxTenor"
      >
        Max Tenor (Days)
      </th>
    </tr>
  </thead>
  <tbody
    className=""
  >
    <tr
      className=""
      id="1"
    >
      <td
        className="left aligned"
        data-test-id="1-checkbox"
        id="1-checkbox"
      >
        <div
          className="ui fitted checkbox"
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <input
            checked={false}
            className="hidden"
            onClick={[Function]}
            readOnly={true}
            tabIndex={0}
            type="checkbox"
          />
          <label />
        </div>
      </td>
      <td
        className=""
        data-test-id="1-source"
        id="1-source"
      >
        <span
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "paddingTop": "8px",
            }
          }
        >
          <span
            className="c0"
          >
            A Company
          </span>
        </span>
      </td>
      <td
        className=""
        data-test-id="1-location"
      >
        <span
          style={
            Object {
              "paddingTop": "8px",
            }
          }
        >
          <span
            className="c1"
          >
            city
          </span>
        </span>
      </td>
      <td
        className=""
        data-test-id="1-appetite"
      >
        <span
          style={
            Object {
              "paddingTop": "8px",
            }
          }
        >
          <span
            className="c1"
          >
            No
          </span>
        </span>
      </td>
      <td
        className=""
        data-test-id="1-availability"
      >
        <span
          style={
            Object {
              "paddingTop": "8px",
            }
          }
        >
          <span
            className="c1"
          >
            Yes
          </span>
        </span>
      </td>
      <td
        className=""
        data-test-id="1-creditLimit"
      >
        <span
          style={
            Object {
              "paddingTop": "8px",
            }
          }
        >
          <span
            className="c1"
          >
            -
          </span>
        </span>
      </td>
      <td
        className=""
        data-test-id="1-riskFee"
      >
        <span
          style={
            Object {
              "paddingTop": "8px",
            }
          }
        >
          <span
            className="c1"
          >
            -
          </span>
        </span>
      </td>
      <td
        className=""
        data-test-id="1-margin"
      >
        <span
          style={
            Object {
              "paddingTop": "8px",
            }
          }
        >
          <span
            className="c1"
          >
            -
          </span>
        </span>
      </td>
      <td
        className=""
        data-test-id="1-maxTenor"
      >
        <span
          style={
            Object {
              "paddingTop": "8px",
            }
          }
        >
          <span
            className="c1"
          >
            -
          </span>
        </span>
      </td>
    </tr>
    <tr
      className=""
      id="2"
    >
      <td
        className="left aligned"
        data-test-id="2-checkbox"
        id="2-checkbox"
      >
        <div
          className="ui fitted checkbox"
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <input
            checked={false}
            className="hidden"
            onClick={[Function]}
            readOnly={true}
            tabIndex={0}
            type="checkbox"
          />
          <label />
        </div>
      </td>
      <td
        className=""
        data-test-id="2-source"
        id="2-source"
      >
        <span
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "paddingTop": "8px",
            }
          }
        >
          <span
            className="c0"
          >
            B Company
          </span>
        </span>
      </td>
      <td
        className=""
        data-test-id="2-location"
      >
        <span
          style={
            Object {
              "paddingTop": "8px",
            }
          }
        >
          <span
            className="c1"
          >
            city
          </span>
        </span>
      </td>
      <td
        className=""
        data-test-id="2-appetite"
      >
        <span
          style={
            Object {
              "paddingTop": "8px",
            }
          }
        >
          <span
            className="c1"
          >
            No
          </span>
        </span>
      </td>
      <td
        className=""
        data-test-id="2-availability"
      >
        <span
          style={
            Object {
              "paddingTop": "8px",
            }
          }
        >
          <span
            className="c1"
          >
            Yes
          </span>
        </span>
      </td>
      <td
        className=""
        data-test-id="2-creditLimit"
      >
        <span
          style={
            Object {
              "paddingTop": "8px",
            }
          }
        >
          <span
            className="c1"
          >
            -
          </span>
        </span>
      </td>
      <td
        className=""
        data-test-id="2-riskFee"
      >
        <span
          style={
            Object {
              "paddingTop": "8px",
            }
          }
        >
          <span
            className="c1"
          >
            -
          </span>
        </span>
      </td>
      <td
        className=""
        data-test-id="2-margin"
      >
        <span
          style={
            Object {
              "paddingTop": "8px",
            }
          }
        >
          <span
            className="c1"
          >
            -
          </span>
        </span>
      </td>
      <td
        className=""
        data-test-id="2-maxTenor"
      >
        <span
          style={
            Object {
              "paddingTop": "8px",
            }
          }
        >
          <span
            className="c1"
          >
            -
          </span>
        </span>
      </td>
    </tr>
    <tr
      className=""
      id="3"
    >
      <td
        className="left aligned"
        data-test-id="3-checkbox"
        id="3-checkbox"
      >
        <div
          className="ui fitted checkbox"
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <input
            checked={false}
            className="hidden"
            onClick={[Function]}
            readOnly={true}
            tabIndex={0}
            type="checkbox"
          />
          <label />
        </div>
      </td>
      <td
        className=""
        data-test-id="3-source"
        id="3-source"
      >
        <span
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "paddingTop": "8px",
            }
          }
        >
          <span
            className="c0"
          >
            C Company
          </span>
        </span>
      </td>
      <td
        className=""
        data-test-id="3-location"
      >
        <span
          style={
            Object {
              "paddingTop": "8px",
            }
          }
        >
          <span
            className="c1"
          >
            city
          </span>
        </span>
      </td>
      <td
        className=""
        data-test-id="3-appetite"
      >
        <span
          style={
            Object {
              "paddingTop": "8px",
            }
          }
        >
          <span
            className="c1"
          >
            No
          </span>
        </span>
      </td>
      <td
        className=""
        data-test-id="3-availability"
      >
        <span
          style={
            Object {
              "paddingTop": "8px",
            }
          }
        >
          <span
            className="c1"
          >
            Yes
          </span>
        </span>
      </td>
      <td
        className=""
        data-test-id="3-creditLimit"
      >
        <span
          style={
            Object {
              "paddingTop": "8px",
            }
          }
        >
          <span
            className="c1"
          >
            -
          </span>
        </span>
      </td>
      <td
        className=""
        data-test-id="3-riskFee"
      >
        <span
          style={
            Object {
              "paddingTop": "8px",
            }
          }
        >
          <span
            className="c1"
          >
            -
          </span>
        </span>
      </td>
      <td
        className=""
        data-test-id="3-margin"
      >
        <span
          style={
            Object {
              "paddingTop": "8px",
            }
          }
        >
          <span
            className="c1"
          >
            -
          </span>
        </span>
      </td>
      <td
        className=""
        data-test-id="3-maxTenor"
      >
        <span
          style={
            Object {
              "paddingTop": "8px",
            }
          }
        >
          <span
            className="c1"
          >
            -
          </span>
        </span>
      </td>
    </tr>
  </tbody>
</table>
`;
